toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/Authenticator.kt

```diff
@@ -185,3 +358,38 @@ private fun Context.findActivity(): Activity {
     }
     throw IllegalStateException("Could not find an activity from which to launch client certificate picker.")
 }
+
+/**
+ * Represents different types of browser-based authentication challenges.
+ *
+ * This sealed class is used to encapsulate specific authentication challenges that require
+ * interaction with a browser.
+ *
+ * @since 200.8.0
+ */
+public sealed class BrowserAuthChallenge {
+    /**
+     * Represents an Identity-Aware Proxy (IAP) sign-in challenge.
+     *
+     * @param iapSignIn The IAP sign-in object containing the necessary configuration.
+     * @since 200.8.0
+     */
+    public data class IapSignInChallenge internal constructor(val iapSignIn: IapSignIn) : BrowserAuthChallenge()
```

Comment:
No need to repeat "Challenge", as it's already in the name of the base class:
```suggestion
    public data class IapSignIn internal constructor(val iapSignIn: IapSignIn) : BrowserAuthChallenge()
```

Reply:
This will not work as it will cause a conflicting declaration since `IapSignIn` is already a type in the API. 
When i change the name, I get the following warning: `Constructor has non-null self reference parameter`